python_apps/rtsp-camera-to-pravega-probe.py

```diff
@@ -0,0 +1,341 @@
+    thread.start()
+    logging.info('Probe server is listening on %s:%d' % (hostname, port))
+    return httpd
+
+class ProbeHttpHandler(BaseHTTPRequestHandler):
+    ts_set = None
+    def send_code_msg(self, code, msg): 
+        self.send_response(code)
+        self.send_header('Content-Type',
+                         'text/plain; charset=utf-8')
+        self.end_headers()
+        self.wfile.write(msg.encode('utf-8'))
+
+    # Any code greater than or equal to 200 and less than 400 indicates success. Any other code indicates failure
+    # https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes/
+    def do_GET(self):
+        if self.path.lower() == "/ishealthy":
+            if ProbeHttpHandler.ts_set.is_uptodate():
+                self.send_code_msg(200, "OK")
+            else:
+                self.send_code_msg(500, "Internal Server Error")
+        else:
+            self.send_code_msg(404, "Not Found")
+
+    def do_POST(self):
+        self.send_code_msg(404, "Not Found")
+
+def main():
+    parser = argparse.ArgumentParser(
+        description="Capture from RTSP camera and write video to a Pravega stream",
+        auto_env_var_prefix="")
+    # Note that below arguments can be passed through the environment such as PRAVEGA_CONTROLLER_URI.
+    parser.add_argument("--allow-create-scope", type=str2bool, default=True)
+    parser.add_argument("--buffer-size-mb", type=float, default=10.0, help='Buffer size in MiB')
+    parser.add_argument("--camera-address")
+    parser.add_argument("--camera-height", type=int, default=180)
+    parser.add_argument("--camera-password")
+    parser.add_argument("--camera-path", default="/")
+    parser.add_argument("--camera-port", type=int, default=554)
+    parser.add_argument("--camera-protocols")
+    parser.add_argument("--camera-rate-KB-per-sec", type=float, default=25.0, help="rate in KB/sec")
+    parser.add_argument("--camera-uri")
+    parser.add_argument("--camera-user")
+    parser.add_argument("--camera-width", type=int, default=320)
+    parser.add_argument("--container-format", default="mp4", help="mpegts or mp4")
+    parser.add_argument("--debugspy", type=str2bool, default=False)
+    parser.add_argument("--fakesink", type=str2bool, default=False)
+    parser.add_argument("--fakesource", type=str2bool, default=False)
+    parser.add_argument("--fragment-duration-ms", type=int, default=1)
+    parser.add_argument("--keycloak-service-account-file")
+    parser.add_argument("--log-level", type=int, default=logging.INFO, help="10=DEBUG,20=INFO")
+    parser.add_argument("--pravega-controller-uri", default="tcp://127.0.0.1:9090")
+    parser.add_argument("--pravega-scope", required=True)
+    parser.add_argument("--pravega-stream", required=True)
+    parser.add_argument("--pravega-buffer-size", type=int, default=1024, help='Pravega writer buffer size in bytes')
+    parser.add_argument("--healthy-seconds", type=int, default=30)
```

Comment:
Keep this list of arguments in alphabetical order.

Reply:
changed accordingly